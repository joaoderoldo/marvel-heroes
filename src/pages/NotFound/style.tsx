import styled from "styled-components";

export const NotFoundContainer = styled.main`
  display: flex;
  justify-content: center;
  align-items: center;
  flex-flow: column wrap;
  height: 100vh;

  img {
    max-width: 500px;
    height: auto;
    object-fit: cover;
    display: block;
  }

  a {
    color: var(--secondary, #ccc);
    font-weight: 600;
    font-size: 1.25rem;
    text-decoration: none;
  }
`;
