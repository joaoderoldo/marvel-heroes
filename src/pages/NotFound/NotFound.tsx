import { NotFoundContainer } from "./style";
import NotFoundImage from "@/assets/img/404.jpg";
import { Link } from "react-router-dom";

const NotFound = () => {
  return (
    <NotFoundContainer>
      <img src={NotFoundImage} alt="Pagina nÃ£o Encontrada" />
      <Link to="/">Voltar</Link>
    </NotFoundContainer>
  );
};

export default NotFound;
