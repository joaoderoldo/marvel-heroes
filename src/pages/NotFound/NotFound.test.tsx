import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";

import NotFound from "./NotFound";

describe("NotFound", () => {
  test("renders correctly", () => {
    const { container } = render(<NotFound />, { wrapper: BrowserRouter });
    expect(container.firstChild).toMatchSnapshot();
  });

  test('renders the "Voltar" link', () => {
    const { getByText } = render(<NotFound />, { wrapper: BrowserRouter });
    const linkElement = getByText(/Voltar/);
    expect(linkElement).toBeInTheDocument();
  });
});
