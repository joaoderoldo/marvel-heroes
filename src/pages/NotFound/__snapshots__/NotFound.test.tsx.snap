// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NotFound > renders correctly 1`] = `
<main
  class="sc-beqWaB dxYkva"
>
  <img
    alt="Pagina nÃ£o Encontrada"
    src="/src/assets/img/404.jpg"
  />
  <a
    href="/"
  >
    Voltar
  </a>
</main>
`;
